saleor/discount/utils.py

```diff
@@ -581,3 +583,21 @@ def get_current_products_for_rules(rules: "QuerySet[PromotionRule]"):
         Exists(rule_variants.filter(productvariant_id=OuterRef("id")))
     )
     return Product.objects.filter(Exists(variants.filter(product_id=OuterRef("id"))))
+
+
+def update_promotion_rules_variants_dirty(is_dirty):
+    rules = get_active_promotion_rules()
+    rules.update(variants_dirty=is_dirty)
+
+
+def get_active_promotion_rules():
+    promotions = Promotion.objects.using(
+        settings.DATABASE_CONNECTION_REPLICA_NAME
+    ).active()
+    return (
+        PromotionRule.objects.order_by("id")
+        .using(settings.DATABASE_CONNECTION_REPLICA_NAME)
```

Comment:
When we are doing update on the queryset I suppose we shouldn't user replica here